litellm/integrations/lunary.py

```diff
@@ -0,0 +1,167 @@
+#### What this does ####
+#    On success + failure, log events to lunary.ai
+from datetime import datetime, timezone
+import traceback
+import dotenv
+import subprocess
+import sys
+
+dotenv.load_dotenv() 
+
+# convert to {completion: xx, tokens: xx}
+def parse_usage(usage):
+    return {
+        "completion": usage["completion_tokens"] if "completion_tokens" in usage else 0,
+        "prompt": usage["prompt_tokens"] if "prompt_tokens" in usage else 0,
+    }
+
+
+def parse_messages(input):
+    if input is None:
+        return None
+
+    def clean_message(message):
+        # if is strin, return as is
+        if isinstance(message, str):
+            return message
+
+        if "message" in message:
+            return clean_message(message["message"])
+       
+        
+        serialized = {
+            "role": message.get("role"),
+            "content": message.get("content"),
+        }
+
+        # Only add tool_calls and function_call to res if they are set
+        if message.get("tool_calls"):
+            serialized["tool_calls"] = message.get("tool_calls")
+        if message.get("function_call"):
+            serialized["function_call"] = message.get("function_call")
+
+        return serialized
+
+    if isinstance(input, list):
+        if len(input) == 1:
+            return clean_message(input[0])
+        else:
+            return [clean_message(msg) for msg in input]
+    else:
+        return clean_message(input)
+
+
+class LunaryLogger:
+    # Class variables or attributes
+    def __init__(self):
+        try:
+            import lunary
+            # lunary.__version__ doesn't exist throws if lunary is not installed
+            if not hasattr(lunary, "track_event"):
+                raise ImportError
+            
+            self.lunary_client = lunary
+        except ImportError:
+            print("Lunary not installed. Installing now...")
+            subprocess.check_call([sys.executable, "-m", "pip", "install", "lunary", "--upgrade"])
```

Comment:
hey @vincelwt we're trying to move away from auto-installs. Can you change this to raise an error to the user, asking them to install your package.

Reply:
Done, thanks!